src/Transformer/DetermineHeroImages.php

```diff
@@ -86,81 +71,88 @@ final class DetermineHeroImages implements Transformer {
 	public function transform( Document $document, ErrorCollection $errors ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
 		$hero_image_elements = [];
 
-		foreach ( [ 'custom_header', 'custom_logo', 'featured_image', 'initial_image_block', 'initial_cover_block' ] as $hero_image_source ) {
+		foreach ( [ 'header_images', 'featured_image', 'initial_content_image' ] as $hero_image_source ) {
 			if ( count( $hero_image_elements ) < PreloadHeroImage::DATA_HERO_MAX ) {
 				$candidate = null;
 
 				switch ( $hero_image_source ) {
-					case 'custom_header':
-						$candidate = $this->get_custom_header( $document );
-						break;
-					case 'custom_logo':
-						$candidate = $this->get_custom_logo( $document );
+					case 'header_images':
+						$candidate = $this->get_header_images( $document );
 						break;
 					case 'featured_image':
 						$candidate = $this->get_featured_image( $document );
 						break;
-					case 'initial_image_block':
-						$candidate = $this->get_initial_content_image_block( $document );
-						break;
-					case 'initial_cover_block':
-						$candidate = $this->get_initial_content_cover_block( $document );
+					case 'initial_content_image':
+						$candidate = $this->get_initial_content_image( $document );
```

Comment:
This feels the riskiest to me that it could  cause an image which is outside the viewport to get loaded non-lazily. But see https://github.com/ampproject/amp-toolbox-php/pull/141#discussion_r610916394.

Reply:
The Twenty Twenty-One Theme has a relatively tall header area where the cover image in the first position doesn't even appear show up:

![image](https://user-images.githubusercontent.com/134745/114260095-9edf4880-9987-11eb-872c-c16aefe945a8.png)

Retaining `loading=lazy` on all content images would guard against this.

Reply:
This is addressed in https://github.com/ampproject/amp-toolbox-php/pull/149. Since content images in WordPress get `loading=lazy`, my update to the `PreloadHeroImage` transformer preserves that on the SSR'ed image. If the original `img` lacked a `loading` attribute (as the Custom Logo does in WP), then the `loading` attribute is likewise omitted from the SSR'ed image.